### ISSUE TITLE
External event starting video

### ISSUE THREAD

**Issue by belforto** [caprica/vlcj#502]
How could I make  video from playlist being started on external event(http request for example)?

**Comment by caprica** [comment]
There are many ways to do this, this question is way too wide in scope.

**Comment by belforto** [comment]
I have video1 and video2 in the playlist.  Goal is to  start video 2  on event over internet (maybe http.get? )while video 1 is looping, and after end of video 2 come back to video 1 looping.

**Comment by caprica** [comment]
So one way is to embed an HTTP web server in your application, or a minimalist REST server and respond to requests.

This sort of thing is really outside the scope of vlcj though.

**Comment by belforto** [comment]
Ok, thanks for the advice. Just which event  from vlcj to implement?

**Comment by caprica** [comment]
I really have no idea what you're asking.

Your web server receives a GET request, inside your GET request handler you invoke mediaPlayer.playMedia(mrl). That's one way to do it. Be wary of threading issues. This is way beyond the scope of this project frankly.

**Comment by belforto** [comment]
Ok I am just thinking about the concept in my head. Will start soon working on it. Thanks for advices

**Comment by caprica** [comment]
You might not even need vlcj. Put your videos on web server. Job done!

**Comment by belforto** [comment]
Maybe but I  need some smooth transition between them. Better would be if they are in some kind of playlist

**Comment by belforto** [comment]
What I did: 1. Created a webapp with servlet.
                   2. Included your test file in sources which was working locally
and got :
 uk.co.caprica.vlcj.Info - vlcj: 3.8.0
Info:   17:07:15.996 [http-listener-1(2)]    INFO                                                   uk.co.caprica.vlcj.Info - java: 1.8.0_45 Oracle Corporation
Info:   17:07:15.997 [http-listener-1(2)]    INFO                                                   uk.co.caprica.vlcj.Info - java home: C:\Program Files\Java\jdk1.8.0_45\jre
Info:   17:07:16.000 [http-listener-1(2)]    INFO                                                   uk.co.caprica.vlcj.Info - os: Windows 8.1 6.3 amd64
Info:   17:07:16.049 [http-listener-1(2)]    DEBUG                             uk.co.caprica.vlcj.discovery.NativeDiscovery - discover()
Info:   17:07:16.049 [http-listener-1(2)]    DEBUG                             uk.co.caprica.vlcj.discovery.NativeDiscovery - jnaLibraryPath=null
Info:   17:07:16.049 [http-listener-1(2)]    DEBUG                             uk.co.caprica.vlcj.discovery.NativeDiscovery - discoveryStrategy=uk.co.caprica.vlcj.discovery.linux.DefaultLinuxNativeDiscoveryStrategy@31842a5c
Info:   17:07:16.049 [http-listener-1(2)]    DEBUG                             uk.co.caprica.vlcj.discovery.NativeDiscovery - supported=false
Info:   17:07:16.050 [http-listener-1(2)]    DEBUG                             uk.co.caprica.vlcj.discovery.NativeDiscovery - discoveryStrategy=uk.co.caprica.vlcj.discovery.windows.DefaultWindowsNativeDiscoveryStrategy@58844efd
Info:   17:07:16.050 [http-listener-1(2)]    DEBUG                             uk.co.caprica.vlcj.discovery.NativeDiscovery - supported=true
Info:   17:07:16.050 [http-listener-1(2)]    DEBUG             uk.co.caprica.vlcj.discovery.AbstractNativeDiscoveryStrategy - discover()
Info:   17:07:16.055 [http-listener-1(2)]    DEBUG                    uk.co.caprica.vlcj.runtime.windows.WindowsRuntimeUtil - getVlcInstallDir()
Warning:   StandardWrapperValve[Marko]: Servlet.service() for servlet Marko threw exception
java.lang.IllegalAccessError: tried to access method com.sun.jna.Pointer.<init>(J)V from class com.sun.jna.platform.win32.WinReg$HKEY
	at com.sun.jna.platform.win32.WinReg$HKEY.<init>(WinReg.java:32)

What is causing the problem? I guess .dll plugin was not loaded, could I include it into project?

**Comment by caprica** [comment]
This looks unrelated to plugins.

Probably mismatched JNA and JNA-platform jars. Are you sure you have the right versions of all of your dependencies? IllegalAccessError should not happen.

**Comment by belforto** [comment]
Yeah seems like some issue while trying to access from jna to jna-platform class, some methods inside are package private others are public I guess.
Could i take some shortcut and locate .dll manually or include it into resources?

**Comment by caprica** [comment]
The .dll plugins issue is not relevant here. An IllegalAccessError is happening because of something like package/class/method access modifiers (public/private/protected). This would ordinarily be picked up at *compile* time, not run time, and by that I mean compile time of JNA, not vlcj or your application.

Clearly there is nothing wrong in this regard with JNA, so I can only assume something like a mismatched JNA/platform jar file is being found on your system.

The solution as such is to debug your classpath.

**Comment by belforto** [comment]
I tried same jna on desktop app everything  is working fine.  When i switch
to web app problems happen.  It might be matter of paths since i tried to
use newest jna and platform combination,  also combine them into one jar
but no effect

Dana 11. 8. 2016. 10:44 osoba "Mark Lee" <notifications@github.com>
napisala je:
